test/filters_test.py

```diff
@@ -201,3 +201,48 @@ def test_fail_wrong_type(self):
       [TryteString(b'RBTC9D9DCDQAEASBYBCCKBFA')],
       [f.Type.CODE_WRONG_TYPE],
     )
+
+
+# noinspection SpellCheckingInspection
+class AddressNoChecksumTestCase(BaseFilterTestCase):
+  filter_type = AddressNoChecksum
+
+  # noinspection SpellCheckingInspection
+  def setUp(self):
+    super(AddressNoChecksumTestCase, self).setUp()
+
+    # Define some addresses that we can reuse between tests
+    """
+    Incoming value is not an :py:class:`Address` instance.
+    """
+    self.tryte1 = (
+      b'TESTVALUE9DONTUSEINPRODUCTION99999FBFFTG'
+      b'QFWEHEL9KCAFXBJBXGE9HID9XCOHFIDABHDG9AHDR'
+    )
+    self.checksum = b'ENXYJOBP9'
+    self.address = Address(self.tryte1)
+    self.address_with_checksum = Address(self.tryte1 + self.checksum)
+    self.address_with_bad_checksum = Address(self.tryte1 + b'DEADBEEF9')
+
+  def test_pass_checksumless_addy(self):
+    """
+    Incoming value is tryte in address form or Address object
+    """
+    self.assertFilterPasses(self.tryte1)
+    self.assertFilterPasses(self.address)
+
+  def test_pass_withchecksum_addy(self):
+    """
+    After passing through the filter an address with a checksum should
+    return the address without
+    """
+    self.assertFilterPasses(self.address_with_checksum, self.address)
+
+  def test_fail_badchecksum_addy(self):
+    """
+    If they've got a bad checksum in their address we should probably tell
+    them so they don't wonder why something works in one place and not another
+    """
+    self.assertFilterErrors(
+      self.address_with_bad_checksum,
+      [AddressNoChecksum.ADDRESS_BAD_CHECKSUM])
```

Comment:
👍 This is a really cool idea.  I like it.

I feel like this feature will be received positively (grudgingly so, perhaps, but positively nonetheless).

If a developer is super confident that they know what they are doing, they can always strip off the checksum before providing it to the API method.  But they have to explicitly bypass the safety mechanism, so it forces the developer to really think about what they're doing (and whether maybe – just maybe – the invalid checksum is because of a bug somewhere else in the code that needs to be addressed).

Reply:
Just a heads up this is ready for review again.